src/inventory/inventory.resolver.ts

```diff
@@ -19,24 +29,94 @@ export class InventoryResolver {
     );
   }
 
-  @Query(() => [ProductItem], { name: 'productItems' })
-  findAll() {
-    return this.inventoryService.findAll();
+  @Query(() => ProductItemConnection, {
+    name: 'productItems',
+    description: 'Retrieves all product items',
+  })
+  async findAll(
+    @Args() args: FindProductItemArgs,
+    @Info() info,
+  ): Promise<IPaginatedType<ProductItem>> {
+    const { first, skip } = args;
+
+    // get query keys to avoid unnecessary workload
+    const query = queryKeys(info);
```

Comment:
is it possible to extract sth here?
this looks really copy-paste

Reply:
What do you propose?
I found the functionality to get the queried proporties and put it in a function

Reply:
idk
it just looked very similar
if you say this is as good as it gets, then I'm fine with that